tools/lister.py

```diff
@@ -66,7 +66,13 @@ def list_files(targets=[], ftypes=[], use_shebang=True, modified_only=False,
     # sys.argv as str, so that battle is already lost.  Settle for hoping
     # everything is UTF-8.
     repository_root = subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).strip().decode('utf-8')
-    exclude_abspaths = [os.path.normpath(os.path.join(repository_root, fpath)) for fpath in exclude]
+    exclude_abspaths = []
+    for path in exclude:
+        normed_path = os.path.normpath(os.path.join(repository_root, path))
+        if os.path.isdir(normed_path):
+            exclude_abspaths += [os.path.normpath(os.path.join(repository_root, fpath)) for fpath in list_files(targets=[normed_path])]
+        else:
+            exclude_abspaths.append(normed_path)
```

Comment:
Are you sure this didn't already work?  This should work:
```
        if any(absfpath == expath or absfpath.startswith(expath + '/')                               
               for expath in exclude_abspaths):                                                      
            continue                                                                                 
```

Reply:
From my tests and the comment in the code, `excludes_paths` didn't exclude directories - we might want to make the same change to `zulip`?

Reply:
I think the comments are inaccurate.  Looking at the history, we used to have `contrib_bots/bots` in the exclude paths in the server repo.

Reply:
Oh right - I see the issue then, it's Windows paths: 
```
if any(absfpath == expath or absfpath.startswith(expath + '/')
               for expath in exclude_abspaths):
            continue
```
hardcodes the delimiter, so it didn't work for the Windows files. Will update that snippet instead.

Reply:
Cool, yeah, and that fix we should contribute to the main repo too.